server/base/src/main/java/org/apache/accumulo/server/util/ManagerMetadataUtil.java

```diff
@@ -199,10 +199,10 @@ public static void replaceDatafiles(ServerContext context, KeyExtent extent,
     TabletMutator tablet = context.getAmple().mutateTablet(extent);
 
     datafilesToDelete.forEach(tablet::deleteFile);
-    scanFiles.forEach(tablet::putScan);
+    scanFiles.stream().map(StoredTabletFile::getTabletFile).forEach(tablet::putScan);
 
     if (path.isPresent()) {
-      tablet.putFile(path.orElseThrow(), size);
+      tablet.putFile(path.orElseThrow().getTabletFile(), size);
```

Comment:
For the putFile method on [TabletMutator](https://github.com/apache/accumulo/blob/0017fd8949cc1bc7253b8e90313d4fd1b71e2241/server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java#L84) I think it would best to add an overloaded version of the method that takes a StoredTabletFile (keeping the one that takes a TabletFile if needed). Then if the code has a StoredTabletFile it can pass it.  The putFile() method is called to add new files and to update the DataFileValue for existing files.  For the case of updating existing files need to pass the StoredTabletFile through instead of converting to the a TabletFile (which normalizes).

Reply:
So that makes sense for updates, but I'm a bit confused as that is not how it works today so I'm wondering what would happen if we change this. 

In the current code we always insert a normalized path in the putFile() method. So even the cases where the instance type is StoredTabletFile, like [here](https://github.com/apache/accumulo/blob/0017fd8949cc1bc7253b8e90313d4fd1b71e2241/server/base/src/main/java/org/apache/accumulo/server/util/ManagerMetadataUtil.java#L205), the inserted path is normalized because we then [call](https://github.com/apache/accumulo/blob/0017fd8949cc1bc7253b8e90313d4fd1b71e2241/server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java#L86) getMetaInsertText() which will use the normalized string and not the metadataEntry string.

Reply:
As I mentioned in the other comment I think these are existing bugs that these changes uncovered and these changes will make it easier to write correct code going forward.

Reply:
Same as my other comment, once #3432 is merged I will update this line to pass in the StoredTabletFile as that PR adds a new overloaded method that will take a StoredTabletFile instead of TabletFile.